zipline/protocol.py

```diff
@@ -194,6 +200,87 @@ def current_portfolio_weights(self):
         return position_values / self.portfolio_value
 
 
+class AlgorithmPortfolio(Portfolio):
+
+    def __init__(self, data_portal, benchmark=None):
+        super(AlgorithmPortfolio, self).__init__()
+        self.data_portal = data_portal
+        self.benchmark = benchmark
+        self.current_date = None
+
+    @property
+    def expected_shortfall(self):
```

Comment:
Noted in person: I would move the lookback length to be a parameter here with a default.